### PR TITLE
OS specifc GitHub Actions

### PR DESCRIPTION
**Pull request by nschonni**
Nothing wrong with the current approach, was just playing with the filtering.
Also noticed that the VScode linter doesn't like the interpolation for the `runs-on` value

### CODE REVIEW

**Comment by nijel** [comment]
This duplicates most of the logic into three files, why do you think it's better than having all in one place?

**Comment by nschonni** [comment]
Some parts at the end are duplicated, but since the matrix don't really overlap nor the setup steps, I (personal opinion) find it easier to follow what is going on in each environment

**Comment by nijel** [comment]
I think the matrix should ideally overlap in the ideal case. There were some issues with Windows builds that made me skip some of the Python versions, but ideally we would test all supported Python versions on Windows as well.

**Comment by nschonni** [comment]
True, but at this point it's only ~50% shared at this point. If the windows testing and all the python versions overlap, it would make sense just leaving the brew/apt as the only branching logic in the file.

I'd also recommend pulling out the "minimal" stuff to its' own job. I didn't really understand what it was doing though so I didn't touch it.

The nice part of the Actions vs. Travis, is that you can split this out, and don't have to shoehorn flow logic into yaml 😄

**Comment by nijel** [comment]
The only difference of minimal job is installing minimal required versions.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -1,31 +1,24 @@
...
 
 jobs:
   test:
 
-    runs-on: ${{ matrix.os }}-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu, windows, macos]
+        os: [ubuntu-latest]
```
Comment (nijel) [inline .github/workflows/linux.yml:17]: Why keeping os in the matrix when there is just one?

File: .github/workflows/win.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Update pip
+      run: pip install --upgrade pip wheel
+    - name: Install pip dependencies
+      run: pip install -r requirements/dev.txt
+    - name: Compile all the files
+      run: python -m compileall -q -f .
+    - name: Setup build
+      run: python setup.py --quiet build
```
Comment (nijel) [inline .github/workflows/win.yml:40]: When doing the cleanup here, those are probably not needed in none of the actions. The setup.py build is tested in separate workflow and it should byte compile the files as well (I guess the reasoning behind this was syntax check, but we do have lint job as well).

```suggestion
```

File: .github/workflows/linux.yml (hunk 1 above)
Comment (nschonni) [reply .github/workflows/linux.yml:17]: This could definitely be inlined, I just kept to avoid updating the caching and coverage keys

Comment (nschonni) [reply .github/workflows/linux.yml]: Done

File: .github/workflows/win.yml (hunk 2 above)
Comment (nschonni) [reply .github/workflows/win.yml]: Interesting, the build looks to be failing after removing this. goin to try and add it back to see if it works again

Comment (nschonni) [reply .github/workflows/win.yml:38]: Adding it back fixed it, but here is the failure when both these steps were removed https://github.com/translate/translate/runs/610261179?check_suite_focus=true#step:8:13 if that rings a bell why

Comment (nschonni) [reply .github/workflows/win.yml]: And failures when only Setup step is removed https://github.com/translate/translate/pull/3981/checks?check_run_id=610316715#step:11:17

Comment (nijel) [reply .github/workflows/win.yml:38]: That's certainly bug in the code, but I don't get why byte-compiling hides it. I've tried to address this in https://github.com/translate/translate/pull/3984

Comment (nschonni) [reply .github/workflows/win.yml:38]: OK, after that lands I can try and remove the steps again

Comment (nijel) [reply .github/workflows/win.yml:38]: I've just merged that.

Comment (nschonni) [reply .github/workflows/win.yml]: Looks like it's catching something else now https://github.com/translate/translate/pull/3981/checks?check_run_id=610959605#step:10:28

Comment (nschonni) [reply .github/workflows/win.yml]: OK, add a suggested fix from pylint. Hope it makes sense

File: translate/misc/selector.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def add(self, path, method_dict=None, prefix=None, **http_methods):
         
         Prefix will override self.prefix for this mapping.
         """
-        # Thanks to S�bastien Pierre 
+        # Thanks to S�bastien Pierre
```
Comment (nijel) [inline translate/misc/selector.py:128]: This seems wrong, encoding issue?

File: translate/misc/selector.py [hunk 4]
```diff
@@ -282,7 +282,7 @@ def _parse_line(self, line, path, methods):
 
 
 class SimpleParser(object):
-    """Callable to turn path expressions into regexes with named groups.
+    r"""Callable to turn path expressions into regexes with named groups.
```
Comment (nijel) [inline translate/misc/selector.py:285]: Seems I've missed this one in https://github.com/translate/translate/pull/3984

File: translate/misc/selector.py (hunk 3 above)
Comment (nijel) [inline translate/misc/selector.py:128]:
```suggestion
        # Thanks to Sébastien Pierre 
```

**Comment by nijel** [comment]
Can you please squash the fix it to your commit?

**Comment by nschonni** [comment]
Done, not sure what the encoding issue was

File: translate/misc/selector.py [hunk 5]
```diff
@@ -125,7 +125,7 @@ def add(self, path, method_dict=None, prefix=None, **http_methods):
         
         Prefix will override self.prefix for this mapping.
         """
-        # Thanks to S�bastien Pierre 
+        # Thanks to Sébastien Pierre
```
Comment (nijel) [inline translate/misc/selector.py:128]:
```suggestion
        # Thanks to Sébastien Pierre
```
Whitespace issue? This should not be on the diff :-)

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**